Source/DataModels/Resources/ResourceCubemap.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "Resource.h"
+#include <memory>
+
+class ResourceTexture;
+
+class ResourceCubemap : virtual public Resource
+{
+public:
+	ResourceCubemap(UID resourceUID,
+		const std::string& fileName,
+		const std::string& assetsPath,
+		const std::string& libraryPath);
+	virtual ~ResourceCubemap() override;
+
+	ResourceType GetType() const override;
+
+	void SaveImporterOptions(Json& meta) override {};
+	void LoadImporterOptions(Json& meta) override {};
+
+	void SaveLoadOptions(Json& meta) override {};
+	void LoadLoadOptions(Json& meta) override {};
+
+	const std::shared_ptr<ResourceTexture>& GetHDRTexture() const;
```

Comment:
I don't know if this is done somewhere else, but we should not return smart pointers by reference. By doing so, we don't increase the reference count, which defeats the purpose of a shared pointer.
```suggestion
	std::shared_ptr<ResourceTexture> GetHDRTexture() const;
```
[This StackOverflow post](https://stackoverflow.com/questions/10643563/how-to-return-smart-pointers-shared-ptr-by-reference-or-by-value) addresses the concerns regarding the cost.

Reply:
i took from reference the resource Skybox that has a vector, i have adapted from there and forgot to delete const and & 👍